lib/services/matchers.go

```diff
@@ -108,7 +109,12 @@ func matchResourceByFilters(resource types.ResourceWithLabels, filter MatchResou
 
 		switch match, err := parser.EvalBoolPredicate(filter.PredicateExpression); {
 		case err != nil:
-			return false, trace.Wrap(err)
+			if utils.HasTraceType(err) {
```

Comment:
Not really a fan of this, especially because that `HasTraceType` depends on us keeping a list of errors up to date with whatever typed errors are added to `trace` (and they're 100% going to get out of sync). Can't we just always return `BadParameter` here? What else can go wrong in `parser.EvalBoolPredicate(filter.PredicateExpression)`?

Reply:
you're right, there are different types of trace that comes up (like `NotFound`) but i don't think there is really any benefit to having it differentiated from `BadParameter`